agent/command/git.go

```diff
@@ -243,12 +245,14 @@ func (c *gitFetchProject) buildCloneCommand(conf *internal.TaskConfig, opts clon
 	if conf.GithubPatchData.PRNumber != 0 {
 		var ref, commitToTest, branchName string
 		if conf.Task.Requester == evergreen.MergeTestRequester {
-			// GitHub creates a ref at refs/pull/[pr number]/merge
-			// pointing to the test merge commit they generate
-			// See: https://developer.github.com/v3/git/#checking-mergeability-of-pull-requests
-			// and: https://docs.travis-ci.com/user/pull-requests/#my-pull-request-isnt-being-built
+			// proceed if github has checked if this pr is mergeable. If it hasn't checked, this request
+			// will make it check. If it's not mergeable (ie. conflict), proceed and let the merge fail
```

Comment:
it'll fail in the merge task. Added a mention of this

Reply:
Ah, I get it now.
Still, why let it continue? It will come out that we're letting all the tasks that run before the merge task run against whatever HEAD of the default branch is. My thinking is that 1) this could be deceptive for users: they think their PR is being tested and 2) we could save execution time by failing all the tasks right away when the PR isn't mergeable.

Reply:
So after https://jira.mongodb.org/browse/EVG-14248 the merge task will run anyway regardless of if this task fails, because we need to send the failure status to github. This gives us more leeway here, but I still think it's more correct to not fail here because that may suggest that a test failed (because there is a failure in a task that has meaning to the user). If the correct course of action is to resubmit, I think having the merge task fail communicates that better

Reply:
Got it. 
I could picture an alternative: if a task encounters a non-mergeable PR it calls an endpoint to dequeue/abort the PR. We could add an [abort info](https://github.com/evergreen-ci/evergreen/blob/85f65fce532a7d8260bd848ad99f2fb59be2f0bb/model/task/task.go#L346) for it and display on the version/task pages.

Reply:
I think that's reasonable, assuming by dequeue/abort you mean to send a status to github (the end state at the moment is that we do abort everything except for the merge task). However I'd still prefer to keep it like this for simplicity. If your main concern is overhead with running the merge task, I think it's a reasonable tradeoff for simplicity because it runs so quickly and infrequently

Reply:
My concern is more about the running the non-merge tasks if we already know the merge isn't going to happen. But in the long term this is probably okay since we won't need to give commit queue tasks such a high priority.

Reply:
Ah I see your point. Coincidentally the code was already exiting early for the non-merge tasks, so I cleaned up some of the text. This doesn't dequeue the version but it does short circuit the task

Reply:
sgtm!